ompi/op/op.h

```diff
@@ -588,15 +601,25 @@ static inline void ompi_op_reduce(ompi_op_t * op, const void *source,
                                           op->o_func.java_data.object);
         return;
     }
-    op->o_func.c_fn(source, target, &count, &dtype);
+    if (0 == (op->o_flags & OMPI_OP_FLAGS_BIGCOUNT)) {
+        op->o_func.c_fn(source, target, &count, &dtype);
+    } else {
+        op->o_func.c_fn_bc(source, target, &full_count, &dtype);
+    }
     return;
 }
 
 static inline void ompi_3buff_op_user (ompi_op_t *op, void * restrict source1, void * restrict source2,
-                                       void * restrict result, int count, struct ompi_datatype_t *dtype)
+                                       void * restrict result, size_t full_count, struct ompi_datatype_t *dtype)
 {
-    ompi_datatype_copy_content_same_ddt (dtype, count, (char*)result, (char*)source1);
-    op->o_func.c_fn (source2, result, &count, &dtype);
+    ompi_datatype_copy_content_same_ddt (dtype, full_count, (char*)result, (char*)source1);
+    if (0 == (op->o_flags & OMPI_OP_FLAGS_BIGCOUNT)) {
+        int count = (int)full_count;  /* protected by loop in only caller of this function */
+        assert(full_count <= INT_MAX);
```

Comment:
This assert may fall prey to UB because it happens after the assignment to `int`, which is only valid if `full_count <= INT_MAX`.